### PR TITLE
JBTM-1922 add REST-AT i18n message codes

### PR DESCRIPTION
**Pull request by swetapatra**
Issue Link:
https://issues.redhat.com/browse/JBTM-1922

RTS
!MAIN !CORE !QA_JTA !QA_JTS_JDKORB !QA_JTS_OPENJDKORB !QA_JTS_JACORB !BLACKTIE !XTS !PERF NO_WIN  !AS_TESTS !TOMCAT !JACOCO !LRA

### CODE REVIEW

**Comment by ochaloup** [comment]
TESTIT

**Comment by mmusgrov** [comment]
I added the `Hold` label while we wait for Sweta to rebase against the the current master branch (5.11.1.Final-SNAPSHOT).

**Comment by ochaloup** [comment]
I'm removing `Hold` label as @swetapatra rebased and she works to fix the compilation failure

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java [hunk 1]
```diff
@@ -0,0 +1,124 @@
+package org.jboss.jbossts.star.util.logging;
+
+import static org.jboss.logging.Logger.Level.ERROR;
+import static org.jboss.logging.Logger.Level.WARN;
+
+import javax.transaction.xa.XAException;
+
+//import javax.json.JsonObject;
```
Comment (mmusgrov) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java:8]: We normally ask that comments provide explanatory information.

**Review by mmusgrov — CHANGES_REQUESTED** [review]
I only had some minor change requests (such as running checkstyle over the changes).
@ochaloup I think the PR looks good (proving my few comments are addressed), what do you think?

File: rts/at/bridge/pom.xml [hunk 2]
```diff
@@ -95,11 +95,18 @@
         </dependency>
 
         <dependency>
-            <groupId>org.jboss.logging</groupId>
-            <artifactId>jboss-logging</artifactId>
-            <version>${version.org.jboss.logging.jboss-logging}</version>
-            <scope>provided</scope>
-        </dependency>
+	      <groupId>org.jboss.logging</groupId>
```
[the remark this answers is not in the document]
Comment (ochaloup) [reply rts/at/bridge/pom.xml:98]: I'm not against this I would like just point that if we put the dependency as default for whole `at` module then all sub-modules will inherit that. It will force the dependency to modules where it's not necessary. For `at` it seems it's just `webservices` module. Still it's a place where such dependency is not needed.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/RESTATLogger.java [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+	import org.jboss.logging.Logger;
+
+
+
+	public final class RESTATLogger {
+	    private RESTATLogger() {
+	    }
+
+	    public static final Logger logger = Logger.getLogger("org.jboss.jbossts.star.util.logging");
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/RESTATLogger.java:12]: I think the category should be here just `org.jboss.jbossts.star`. Other Narayana loggers do not use any suffix like here `util.logging`, plus I don't think it makes sense to say all log messages are util ones.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+public interface at18NLogger {
+	
+
+    /*
+        Message IDs are unique and non-recyclable.
+        Don't change the purpose of existing messages.
+          (tweak the message text or params for clarification if you like).
+        Allocate new messages by following instructions at the bottom of the file.
+     */
+	@LogMessage(level = ERROR)
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java:28]: indentation typo?

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    @Message(id = 27001, value = "REST-AT transaction is required for this request. '%s'")
+    void error_rest_at_transactionrequired(int statusCode, @Cause Throwable t);
+
+    @LogMessage(level = ERROR)
+    @Message(id = 27002, value = "REST-AT transaction is not supported by this resource. '%s'")
+    void error_rest_at_transactionnotsupported(int statusCode,@Cause Throwable t);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java:34]: it would be better to have `space` after the comma (`,`). The REST AT module does work with a checkstyle but this is usual. Not necessary to change, but nice to have.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+
+    /*
+        Message IDs are unique and non-recyclable.
+        Don't change the purpose of existing messages.
+          (tweak the message text or params for clarification if you like).
+        Allocate new messages by following instructions at the bottom of the file.
+     */
+	@LogMessage(level = ERROR)
+    @Message(id = 27001, value = "REST-AT transaction is required for this request. '%s'")
+    void error_rest_at_transactionrequired(int statusCode, @Cause Throwable t);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java:30]: please, remove the `_rest_at` prefix at all the method names in this class. Any use of them is prefixed with the `at18NLogger` and it's obvious they're part of this REST AT logging class. Having every method prefixed this  way just brings naming duplicity.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java (hunk 5 above)
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java:34]: It seems methods `error_rest_at_transactionrequired` and `error_rest_at_transactionnotsupported` are not used. If it's so then please just remove them and renumber the ids.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    void error_rest_at_systemexception(String cause);
+    
+    @LogMessage(level = WARN)
+    @Message(id = 27008, value = "Object Store Exception. '%s'")
+    void warn_rest_at_objectStoreException(String cause, @Cause Throwable t);
+    
+    
+    @LogMessage(level = WARN)
+    @Message(id = 27009, value = "IOException. '%s'")
+    void warn_rest_at_IOException(String cause, @Cause Throwable t);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java:68]: I understand that this message was probably taken from the message used before in the code. But message which says "IOException" and then puts "IOException" to the details of the log (caused by `@Cause` annotation) has no much sense. It would be good if the message (and the name of the method) could be change to say like "cannot read suoradinate atomic action from recovery store" or something similar. Plus, using the `e.getMessage()` as message when the log contains whole stacktrace (again the `@Cause` is used here) is ambiguous - the `e.getMessage` will be printed twice without no good reason.

The same points are true for `warn_rest_at_objectStoreException` method.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java [hunk 8]
```diff
@@ -671,6 +675,7 @@ public static int getIntValue(String content, String name, int defValue)
             try {
                 return Integer.parseInt(v);
             } catch (NumberFormatException e) {
+            	RESTATLogger.at18NLogger.warn_rest_at_txsupportGetIntValue(e.getMessage(),e);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java:678]: indentation typo?

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java [hunk 9]
```diff
@@ -482,6 +483,8 @@ public String httpRequest(int[] expect, String url, String method, String mediaT
                 body = (status != -1 ? getContent(connection) : "");
             } catch (IOException e) {
                 body = "";
+                RESTATLogger.at18NLogger.warn_rest_at_txSupportHttpRequest(e.getMessage(),e);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java:486]: I'm not sure but here it seems the exception is an "expected state". There should not be logged warning as until now the exception was just ignored. I think it's good to not hid exception but I recommend to use `debug` or `trace` level but not warning.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java [hunk 10]
```diff
@@ -547,6 +550,7 @@ private void addLocationHeader(HttpURLConnection connection, Map<String, String>
             if (connection.getResponseCode() == HttpURLConnection.HTTP_CREATED)
                 links.put("location", connection.getHeaderField("location"));
         } catch (IOException e) {
+        	RESTATLogger.at18NLogger.warn_rest_at_txSupportAddLocationHeader(e.getMessage(),e);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java:553]: similar to what I said for `warn_rest_at_txSupportHttpRequest` - this seems to be "expected error" state. The logging of the exception should go with `debug` or `trace` but not with warning.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/exception/GenericATException.java [hunk 11]
```diff
@@ -0,0 +1,31 @@
+package org.jboss.jbossts.star.util.exception;
+
+import javax.ws.rs.WebApplicationException;
+import java.net.URL;
+public class GenericATException extends WebApplicationException {
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/exception/GenericATException.java:5]: I don't understand what's the purpose of creating the `GenericATException`. I can't find the usage in the code. Is it meant as some SDK?
Can you please elaborate what is that meant for? Thanks.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java [hunk 12]
```diff
@@ -0,0 +1,124 @@
...
+import org.jboss.logging.annotations.Cause;
+import org.jboss.logging.annotations.LogMessage;
+import org.jboss.logging.annotations.Message;
+import org.jboss.logging.annotations.MessageLogger;
+
+/**
+ * at18N log messages for the rest-at module.
+ */
+@MessageLogger(projectCode = "AT")
+public interface at18NLogger {
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java:19]: please rename the class to `atI18NLogger`

**Review by ochaloup — CHANGES_REQUESTED** [review]
I think it's a great work and I like the PR.
I'm just adding few comments to improve the work.

Then, I'm not sure if it's intentional, if it was not in scope but I found use of `LOG.warn` calls.
Namely in these classes
```
org.jboss.narayana.rest.bridge.inbound.InboundBridgeParticipantDeserializer
org.jboss.narayana.rest.integration.RecoveryManager
org.jboss.narayana.rest.integration.ParticipantResource
org.jboss.narayana.rest.integration.VolatileParticipantResource
org.jboss.jbossts.star.service.Coordinator
org.jboss.jbossts.star.service.TMApplication
org.jboss.jbossts.star.resource.RecoveringTransaction
```
Is there a plan to address those warn calls for `i18N logger`?

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/RESTATLogger.java (hunk 3 above)
Comment (ochaloup) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/RESTATLogger.java:12]: And the name of the variable `at18NLogger` would be probably good to be renamed to `atI18NLogger` or maybe just to `i18NLogger` as using variable with name `i18NLogger` is a usual in other Narayana logger classes.

File: rts/at/util/pom.xml [hunk 13]
```diff
@@ -88,13 +88,19 @@
...
       </exclusions>
     </dependency>
-
-    <dependency>
-      <groupId>org.jboss.logging</groupId>
-      <artifactId>jboss-logging</artifactId>
-      <version>${version.org.jboss.logging.jboss-logging}</version>
-      <scope>provided</scope>
-    </dependency>
+ <!-- jboss logging -->
```
[the remark this answers is not in the document]
Comment (mmusgrov) [reply rts/at/util/pom.xml:91]: The style warnings will be covered by my comment to apply checkstyle to the PR.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/RESTATLogger.java (hunk 3 above)
Comment (mmusgrov) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/RESTATLogger.java:12]: +1 to both comments

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java (hunk 6 above)
Comment (mmusgrov) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java:30]: I thought about that too and agree that it makes sense.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java (hunk 9 above)
Comment (mmusgrov) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java:486]: +1

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java (hunk 10 above)
Comment (mmusgrov) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java:553]: The method is adding the Location Header to the map that the client passed in. The client may well be relying on the Location header to be copied into the map. So I think the original code should have logged a warning here and Sweta's addition of the warning is necessary.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java (hunk 12 above)
Comment (mmusgrov) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java:19]: +1

Comment (swetapatra) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java]: Done

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/exception/GenericATException.java (hunk 11 above)
Comment (swetapatra) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/exception/GenericATException.java]: Removed as no use right now

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java (hunk 10 above)
Comment (swetapatra) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java]: Done

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java (hunk 9 above)
Comment (swetapatra) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/TxSupport.java]: Changed to DEBUG

File: rts/at/util/pom.xml (hunk 13 above)
[the remark this answers is not in the document]
Comment (swetapatra) [reply rts/at/util/pom.xml]: Done

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/exception/GenericATException.java [hunk 14]
```diff
@@ -0,0 +1,31 @@
...
+	    /**
+	     * Generic AT exception.
+	     *
+	     * @param statusCode  error status code behind the creation of this exception
+	     * @param message  error exception message
+	     * @param cause  cause of why this exception is created
+	     */
+	    public GenericATException(int statusCode, String message, Throwable cause) {
+	        super( message, cause);
+
```
[the remark this answers is not in the document]
Comment (swetapatra) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/exception/GenericATException.java]: Done

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/RESTATLogger.java (hunk 3 above)
Comment (swetapatra) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/RESTATLogger.java]: Changed naming

File: rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java (hunk 4 above)
Comment (swetapatra) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/util/logging/at18NLogger.java]: Done

File: rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/InboundBridgeRecoveryModule.java [hunk 15]
```diff
@@ -146,7 +147,7 @@ public void periodicWorkSecondPass() {
                 } while (!finished);
             }
         } catch (Exception e) {
-            LOG.warn(e.getMessage(), e);
+        	RESTATLogger.atI18NLogger.warn_InboundBridgeRecoveryModulegetUidsToRecover(e.getMessage(), e);
```
Comment (mmusgrov) [inline rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/InboundBridgeRecoveryModule.java:150]: checkstyle reports the use of the tabs as errors (so when I enable the plugin my IDE reports that the file as having errors).

Comment (ochaloup) [reply rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/InboundBridgeRecoveryModule.java:150]: +1

File: rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java [hunk 16]
```diff
@@ -0,0 +1,183 @@
...
+	@LogMessage(level = WARN)
+    @Message(id = 27001, value = "Failure while removing participant information from the object store. '%s'")
+    void warn_failureRemovingParticipantObjectStore(String cause, @Cause Throwable t);
+    
+    @LogMessage(level = WARN)
+    @Message(id = 27002, value = "Failure while synchronizing participant url with RecoveryManager. '%s'")
+    void warn_synchronizeParticipantUrlWithCoordinatorRecoveryManager(String cause, @Cause Throwable t);
+    @LogMessage(level = WARN)
+    @Message(id = 27003, value = "Failed to start the bridge. '%s'")
+    void warn_failedtostartbridge(String response, @Cause Throwable t);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java:40]: camel case?

File: rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java [hunk 17]
```diff
@@ -0,0 +1,183 @@
...
+    
+    @LogMessage(level = WARN)
+    @Message(id = 27004, value = "Failed to stop the bridge. '%s'")
+    void warn_failedtostopbridge(String response, @Cause Throwable t);
+    
+    @LogMessage(level = WARN)
+    @Message(id = 27005, value = "Failed to enlist inbound bridge to the transaction '%s'")
+    void warn_failedtoenlisttransaction(String response, @Cause Throwable t);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java:48]: camel case?

File: rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java [hunk 18]
```diff
@@ -0,0 +1,183 @@
...
+    
+    @LogMessage(level = WARN)
+    @Message(id = 27006, value = "Failed to import transaction. '%s'")
+    void warn_failedtoimportTransaction(String response, @Cause Throwable t);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java:52]: camel case?

File: rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java [hunk 19]
```diff
@@ -0,0 +1,183 @@
...
+        Allocate new messages by following instructions at the bottom of the file.
+     */
+	
+	@LogMessage(level = WARN)
+    @Message(id = 27001, value = "Failure while removing participant information from the object store. '%s'")
+    void warn_failureRemovingParticipantObjectStore(String cause, @Cause Throwable t);
+    
+    @LogMessage(level = WARN)
+    @Message(id = 27002, value = "Failure while synchronizing participant url with RecoveryManager. '%s'")
+    void warn_synchronizeParticipantUrlWithCoordinatorRecoveryManager(String cause, @Cause Throwable t);
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java:37]: an empty line could be added down here

File: rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java [hunk 20]
```diff
@@ -0,0 +1,183 @@
...
+
+import static org.jboss.logging.Logger.Level.ERROR;
+import static org.jboss.logging.Logger.Level.WARN;
+import static org.jboss.logging.Logger.Level.DEBUG;
+
+import javax.transaction.xa.XAException;
+
+import org.jboss.logging.Logger.Level;
+
+//import javax.json.JsonObject;
```
Comment (ochaloup) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java:11]: remove this line with comment here please

File: rts/at/tx/src/main/java/org/jboss/jbossts/star/service/TMApplication.java [hunk 21]
```diff
@@ -68,8 +69,7 @@ public TMApplication(Class<?> ... extraClasses)
             Collections.addAll(classes, extraClasses);
         }
         catch (Throwable e)
-        {
-            log.warnf("TM JAX-RS application failed to start: %s", e.getMessage());
+        {   RESTATLogger.atI18NLogger.warn_jaxrsTM(e.getMessage(),e);
```
Comment (ochaloup) [inline rts/at/tx/src/main/java/org/jboss/jbossts/star/service/TMApplication.java:72]: the log statement would be good to be on the separate line from the `{`

**Review by ochaloup — COMMENTED** [review]
I really like it. I have just a minor comments about style (plus the point about the checkstyle). Otherwise perfect.

File: rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/InboundBridge.java [hunk 22]
```diff
@@ -101,10 +103,10 @@ public void start() {
         }
 
         try {
-            TransactionManager.transactionManager().resume(transaction);
+        	TransactionManager.transactionManager().resume(transaction);
```
Comment (mmusgrov) [inline rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/InboundBridge.java:106]: I would avoid tabs and favour of space characters instead making sure there are four of them (unless the existing file already makes extensive use of tabs which is not the case here). If you enable the checkstyle plugin or enable it in your IDE using the wildfly checkstyle rules (as described in the link I gave in my previous review) then these errors/warnings will be hightlighted.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/logging/RESTATLogger.java [hunk 23]
```diff
@@ -0,0 +1,16 @@
+package org.jboss.jbossts.star.logging;
```
[the remark this answers is not in the document]
Comment (mmusgrov) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/logging/RESTATLogger.java:23]: The new copyright format is:

```
Copyright Red Hat

SPDX-License-Identifier: LGPL-2.1-only
```

File: rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/TransactionalExceptionMapper.java [hunk 24]
```diff
@@ -13,18 +13,15 @@
 @Provider
 public class TransactionalExceptionMapper implements ExceptionMapper<TransactionalException> {
 
-    public static final String TRANSACTIONA_REQUIRED_MESSAGE = "REST-AT transaction is required for this request.";
```
Comment (mmusgrov) [inline rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/TransactionalExceptionMapper.java:16]: Why did you move these constants into the EJBExceptionMapper class?

File: rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/InboundBridgeRecoveryModule.java (hunk 15 above)
Comment (mmusgrov) [reply rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/InboundBridgeRecoveryModule.java:150]: It is also possible to run the checkstyle plugin directly. Our top level pom includes the maven-checkstyle-plugin. But when I look at that it refers to `<configLocation>wildfly-checkstyle/checkstyle.xml</configLocation>` and we don't have that file in our repo.

The config is also missing the wildfly-checkstyle/suppressions.xml suppressions filter: `https://maven.apache.org/plugins/maven-checkstyle-plugin/examples/suppressions-filter.html`

@ochaloup have you ever run checkstyle from the maven command?

Comment (mmusgrov) [reply rts/at/bridge/src/main/java/org/jboss/narayana/rest/bridge/inbound/InboundBridgeRecoveryModule.java:150]: @ochaloup I forgot that by default we disable the checkstyle plugin (because much arjuna code dates back many years and we do not want change the orginal code. So the maven command to run the check is:

`mvn checkstyle:checkstyle -Dcheckstyle.skip=false`

and the output will appear in the console.

**Review by ochaloup — APPROVED** [review]
By me it seems good. If the CI pass I will merge this.
Thanks @swetapatra for your work!

**Comment by ochaloup** [comment]
@mmusgrov I think all points were addressed. I'm going to merge.

We can change in a follow-up PR if something is missing.
Maybe there is a question about header style. I advised to use it in this form but you may have a different idea. I can change it when it's clear how it should be styled.

File: rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java [hunk 25]
```diff
@@ -0,0 +1,203 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2021, Red Hat, Inc.
+ *
+ * SPDX-License-Identifier: LGPL-2.1-only
+ *
+ * This is free software; you can redistribute it and/or modify it
```
Comment (mmusgrov) [inline rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java:7]: @ochaloup do you know why this extra text was added?

Comment (ochaloup) [reply rts/at/util/src/main/java/org/jboss/jbossts/star/logging/atI18NLogger.java:7]: Yes, the text was added as I was not clear how form the header should have. We established it this way. If that was wrong please create a PR when you'll be adding the information into README. Sorry for the inconveniences if there is some.
